packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts

```diff
@@ -339,7 +344,60 @@ export class ChainAdapter implements IChainAdapter<ChainTypes.Bitcoin> {
     return { valid: false, result: chainAdapters.ValidAddressResultType.Invalid }
   }
 
-  async subscribeTxs(): Promise<void> {
-    throw new Error('Not implemented')
+  async subscribeTxs(
+    input: chainAdapters.SubscribeTxsInput,
+    onMessage: (msg: chainAdapters.SubscribeTxsMessage<ChainTypes.Bitcoin>) => void,
+    onError: (err: chainAdapters.SubscribeError) => void
+  ): Promise<void> {
+    const {
+      wallet,
+      bip32Params = ChainAdapter.defaultBIP32Params,
+      scriptType = BTCInputScriptType.SpendWitness
+    } = input
+
+    const { xpub } = await this.getPubKey(wallet, bip32Params, scriptType)
+    const account = await this.getAccount(xpub)
+    const addresses = (account.chainSpecific.addresses ?? []).map((address) => address.pubkey)
+
+    await this.providers.ws.subscribeTxs(
+      { topic: 'txs', addresses },
+      (msg) => {
+        const status =
+          msg.confirmations > 0 ? chainAdapters.TxStatus.Confirmed : chainAdapters.TxStatus.Pending
+
+        const baseTx = {
+          address: msg.address,
+          asset: ChainTypes.Bitcoin,
+          blockHash: msg.blockHash,
+          blockHeight: msg.blockHeight,
+          blockTime: msg.blockTime,
+          chain: ChainTypes.Bitcoin as ChainTypes.Bitcoin,
```

Comment:
?

Reply:
fixed this type error:
```
Types of property 'chain' are incompatible.
      Type 'ChainTypes' is not assignable to type 'ChainTypes.Bitcoin'.
```

Reply:
I can type `baseTx` to get around it as well, but this was less lines, is uggo though..